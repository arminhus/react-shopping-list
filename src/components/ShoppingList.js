import React from "react";

export default function ShoppingList(props) {
  return (
    <li>
      {props.items.map((item, i) => {
        <ShoppingList key={i} item={item} />;
      })}
    </li>
  );
}

ShoppingList.defaultProps = {
  items: []
};
