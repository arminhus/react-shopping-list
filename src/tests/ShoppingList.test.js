import React from "react";
import Enzyme from "enzyme";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import ShoppingList from "../components/ShoppingList";

Enzyme.configure({ adapter: new Adapter() });
describe("ShoppingList component test", () => {
  test("should render empty give no items", () => {
    const wrapper = shallow(<ShoppingList />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test("should render ShoppingItems when supplied with an items prop", () => {
    const testItems = [
      { name: "test item 1", checked: false },
      { name: "another test item", checked: true },
      { name: "test item number three", checked: false }
    ];
    const wrapper = shallow(<ShoppingList items={testItems} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
