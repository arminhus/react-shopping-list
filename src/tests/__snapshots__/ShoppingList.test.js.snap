// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShoppingList component test should render ShoppingItems when supplied with an items prop 1`] = `<li />`;

exports[`ShoppingList component test should render empty give no items 1`] = `<li />`;
